NU6.1 final specification changes

daira opened the pull request:
## ZIP 214
* Revision 2: Set the `FS_FPF_ZCG_H3` Mainnet address (same as Revision 1).

## ZIP 255
* Set NU6.1 Mainnet activation height.

## ZIP 271
* Simplify the definition of $\mathsf{ZIP271DisbursementAmount}$ to reflect the fact that NU6.1 will activate on Mainnet at the expiry of the ZIP 214 Revision 1 lockbox stream.
* Require the disbursement to be split into $\mathsf{ZIP271DisbursementChunks}$ chunks.
* Add the Mainnet and Testnet addresses.
* Add the Mainnet BIP 388 wallet descriptor.
* Add Deployment and Reference implementation sections.
* Add a link to this pull request.
* Add links to zcash/zcash#7059 and zcash/librustzcash#1979 in the Reference implementation section.
* Fix markup errors.

## Protocol spec
* Apply new changes from ZIP 271.
* Remove "proposal" from the version string.
* Release version 2025.6.1.

daira committed:
ZIP 255: Set the NU6.1 activation height.

Signed-off-by: Daira-Emma Hopwood <daira@jacaranda.org>

daira committed:
ZIP 271:
* Simplify the definition of ZIP271DisbursementAmount to reflect the fact
  that NU6.1 will activate on Mainnet at the expiry of the ZIP 214
  Revision 1 lockbox stream.
* Add Deployment and Reference implementation sections.
* Fix markup errors.

Signed-off-by: Daira-Emma Hopwood <daira@jacaranda.org>

arya2 reviewed (commented):
These changes look good, if they're the final spec changes for NU6.1, they should also include the:
- Mainnet revision 2 address for FPF (same as the address on Mainnet for revision 1) in ZIP 214, and
- recipient address for the one-time lockbox disbursement on Mainnet (I can't find where the one for Testnet was specified, we may need to add that too).

nuttycom commented on zips/zip-0271.md:
```diff
@@ -450,6 +435,18 @@ As with Option 2a, key compromise or loss would require an emergency Network
+This proposal will be deployed with the NU6.1 network upgrade. [^zip-0255]
+
+
+# Reference implementation
+
+* [zcashd PR 7032: Network Upgrade 6.1](https://github.com/zcash/zcash/pull/7032)
```
```suggestion
# Reference implementation

* [zcashd PR 7032: Network Upgrade 6.1](https://github.com/zcash/zcash/pull/7032)
* [zcashd PR 7059: Release zcashd v6.10.0](https://github.com/zcash/librustzcash/pull/1979)
```

nuttycom reviewed (dismissed):
ACK with suggestion

str4d commented on zips/zip-0271.md:
```diff
@@ -76,23 +76,8 @@ Note: The value $\mathsf{ZIP271DisbursementAmount}$ needs to be precalculated
-($1875\_0000$ zatoshi) is the per-block contribution to the deferred pool during the
-current dev fund.
+$\mathsf{ZIP271DisbursementAmount}$ MUST be $78750\_0000\_0000$ zatoshi on both
+Testnet and Mainnet (i.e. 78,750 TAZ on Testnet, 78,750 ZEC on Mainnet).
```
Now that we have decided the split is 10 equal-sized chunks, we need to specify that, because it's what the consensus rules are checking for (a miner can't arbitrarily choose the number of chunks, like the current wording would suggest).

Specifically, this sentence needs fixing:
```
The additional coinbase output(s) MUST pay a sum of $\mathsf{ZIP271DisbursementAmount}$
zatoshi to the above P2SH multisig address,
```

Additionally, "the above P2SH multisig address" does not exist. We need to specify it, and should also include the corresponding BIP 388 wallet policy so that anyone can derive all of the P2SH addresses that might be used by the KHO's transactions:
```
Wallet descriptor template: sh(sortedmulti(2,@0/**,@1/**,@2/**))
Key information vector:
[2efd03f5/48'/133'/0'/133000']xpub6ERFr6gbA9jHXGj9Qxebua1Fj78ADQJPMyFgP78rUz9kdr4kMMXjkbGQSk6GPD8aYsyPPDXUqftNbuW3jKcHPP1hBa1paPv9FNMLqCqWsNz
[281d1aeb/48'/133'/0'/133000']xpub6FNah9qMu1jMRQdsFXXYWfWXXMhsdP3KRAexTBnnMykskq7VWQGm4439zUwAxNPGVfi7nuL5UWhLwGrHmfnxvLFEx3coS2tjYrezDavvh3T
[e1c4c104/48'/133'/0'/133000']xpub6E2qmBdHpQu9yJ3HvYJ8p8FVYMEWG4a6St2Knd7vfQC9jm7gyWZdvxMo6FafcKvjxiSmd61NW2Mu7udKDbcA5DAVzVdNU6JPrY2uhtPXxE4
```

str4d reviewed (dismissed):
Reviewed 7a04590afb20fd52949533c0c61059168d98b952.

daira committed:
ZIP 271 and protocol spec — address review comments by @str4d:
* The one-time disbursement must be split into 10 chunks.
* The disbursement addresses must be specified.
* Reference ZIP 48.
* Include the BIP 388 wallet policy.

Signed-off-by: Daira-Emma Hopwood <daira@jacaranda.org>

daira committed:
ZIP 271: add a link to this pull request.

Signed-off-by: Daira-Emma Hopwood <daira@jacaranda.org>

daira committed:
Protocol spec: remove "proposal" from the version string and set the
Change History entry date.

Signed-off-by: Daira-Emma Hopwood <daira@jacaranda.org>

daira committed:
ZIP 271: add links to https://github.com/zcash/zcash/pull/7059 and
https://github.com/zcash/librustzcash/pull/1979 in the Reference
implementation section.

Co-authored-by: Kris Nuttycombe <kris@nutty.land>
Signed-off-by: Daira-Emma Hopwood <daira@jacaranda.org>

daira commented:
Ready for re-review.

daira committed:
ZIP 214: set the `FS_FPF_ZCG_H3` Mainnet address (same as the one for
Revision 1).

Signed-off-by: Daira-Emma Hopwood <daira@jacaranda.org>

arya2 reviewed (approved):
ACK 82d2cfa524a2262cc212e08fb92f9f28f5e7e9db

nuttycom reviewed (approved):
ACK 82d2cfa524a2262cc212e08fb92f9f28f5e7e9db

str4d reviewed (approved):
ACK 82d2cfa524a2262cc212e08fb92f9f28f5e7e9db with @aphelionz
